domino_data/data_sources.py

```diff
@@ -316,6 +341,15 @@ def http(self) -> httpx.Client:
             self._httpx = httpx.Client(verify=context)
         return self._httpx
 
+    def pool_manager(self) -> urllib3.PoolManager:
+        """Urllib3 pool manager for range downloads."""
+        if self.datasource_type == DatasourceDtoDataSourceType.ADLSCONFIG.value:
+            return urllib3.PoolManager(headers=ADLS_HEADERS)
+        elif self.datasource_type == DatasourceDtoDataSourceType.GENERICS3CONFIG.value:
+            return urllib3.PoolManager(assert_hostname=False)
```

Comment:
are these things you tested out and figured that they are required given datasource type?

Reply:
it's the urllib3 equivalent of https://github.com/dominodatalab/domino-data/blob/main/domino_data/data_sources.py#L304